python/TestHarness/TestHarness.py

```diff
@@ -357,16 +357,32 @@ def findAndRunTests(self, find_only=False):
         self.preRun()
         self.start_time = clock()
         launched_tests = []
+
+        #used for -i
         if self.options.input_file_name != '':
+
             self._infiles = self.options.input_file_name.split(',')
 
-        if self.options.spec_file and os.path.isdir(self.options.spec_file):
-            search_dir = self.options.spec_file
-        elif self.options.spec_file and os.path.isfile(self.options.spec_file):
-            search_dir = os.path.dirname(self.options.spec_file)
-            self._infiles = [os.path.basename(self.options.spec_file)]
+            if(os.path.dirname(self._infiles[0]) !=''):
+                print('ERROR entered a path, -i can only accept a file name or directory name.')
+                sys.exit(1)
+
+        #used for --run-dir
```

Comment:
You don't need comments like this. They are self evident from the line below. I know you are following the pattern already established. If anything, they should all be stripped unless they are explaining something. No need to state things that can be verified on simple inspection.